driver/registry_base.go

```diff
@@ -239,7 +239,10 @@ func (m *RegistryBase) KeyCipher() *jwk.AEAD {
 
 func (m *RegistryBase) CookieStore() sessions.Store {
 	if m.cs == nil {
-		m.cs = sessions.NewCookieStore(m.C.GetCookieSecrets()...)
+		cs := sessions.NewCookieStore(m.C.GetCookieSecrets()...)
+		cs.MaxAge(0)
```

Comment:
Setting max-age to zero means that the session will be deleted from the browser when you close the browser. I don't think this is the intention here?

Reply:
Nope, definitely not. But the session is actually not deleted on browser restart.

In fact, setting it to zero here has no effect on client side `oauth2_authentication_session` cookie expiration date, as it's being overwritten anyway, in case user chooses to remember session:
https://github.com/ory/hydra/blob/2287ac592fc5a06b1bd3e9f54340a67395452b46/consent/strategy_default.go#L465
And the only place `MaxAge` is used at is inside secure cookie validation then:
https://github.com/gorilla/securecookie/blob/f37875ef1fb538320ab97fc6c9927d94c280ed5b/securecookie.go#L339

But `oauth2_authentication_csrf` and `oauth2_consent_csrf` will have zero epiration though, as they rely on the default value set by `CookieStore` and are not overwritten:
https://github.com/ory/hydra/blob/2287ac592fc5a06b1bd3e9f54340a67395452b46/consent/helper.go#L66
So if someone decides to resume oauth flow on browser restart, it will break it.

Should we leave `cs.MaxAge(0)` here and consider overriding default `MaxAge` for csrf cookies?
Another alternative I thought about is adding a config `serve.cookies.max_age` with default value set to `2592000` (86400 * 30) and use it here `cs.MaxAge(max_age)` to be consistent with current behavior. And then let users override it from configuration and set to any value greater than 30 days if needed. It will extend expiration of csrf cookies as well though, not sure if it can be a problem.

Or would you recommend a better approach to the issue overall?

Reply:
Oh I see, nice find. Not sure how we can test that though but as long as e2e tests pass it should just work. I think your patch is sufficient then but I would probably link in a codedoc to this comment for context!

Reply:
Great! Would you be able to help with this part or point me to where in the docs you'd like to have it mentioned?

Reply:
Added a comment in the code with short summary and a reference to more detailed explanation here.

Reply:
perfect!